hibernate-core/src/main/java/org/hibernate/type/EntityType.java

```diff
@@ -708,4 +725,15 @@ public Object loadByUniqueKey(
 		return result == null ? null : persistenceContext.proxyFor( result );
 	}
 
```

Comment:
The error message below refers to "many-to-one", and I don't think it makes sense for a ```OneToOneType``` attribute. 

Also, ```EntityType#getLHSPropertyName``` returns ```null```, so it's not informative.

I'm not sure of the point of having ```#requireIdentifierOrUniqueKeyType```, since ```#getIdentifierOrUniqueKeyType``` is supposed to throw ```MappingException``` if the ```Type``` can't be resolved. Also, There are lots of usages of ```#getIdentifierOrUniqueKeyType``` that assume the return value is not ```null```. If ```#getIdentifierOrUniqueKeyType``` returned ```null```, I think we'd be seeing ```NullPointerException```.

I'm ok with removing  ```#requireIdentifierOrUniqueKeyType``` and changing references to that method to ```#getIdentifierOrUniqueKeyType```, but please check with @sebersole in case I'm missing something here.